### ISSUE TITLE
Chat - After switching account from Spanish to English the app still shows in Spanish and no option to change language for gmail accounts

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#5286]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Launch the app
2. Log in with an account which is Spanish  (expensifail domain account)
3. Log out from account
4. Log in with an account which is English  (Expensifail or gmail account)
5. Go to Settings
6. Check the language of the app
## Expected Result:
After switch accounts from Spanish to English, all Setting in English language displayed
## Actual Result:
After switch accounts from Spanish to English, all Setting in Spanish language displayed. For public domain accounts there is no option to switch the language.
## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.0.99-0
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/133536100-be7397b6-c288-44a2-84af-5232c32e6439.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @bondydaa (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by francoisl** [comment]
I can reproduce this on production, so I don't think it needs to be a deploy blocker? Going to test again with different accounts to make sure.

**Comment by francoisl** [comment]
I can reproduce in production, so I'm going to remove the deploy blocker label and we can make this a regular issue to fix.

**Comment by bondydaa** [comment]
thanks for taking a look Franc, tossing to the pool

**Comment by MelvinBot** [comment]
Triggered auto assignment to @Christinadobrzyn (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by Christinadobrzyn** [comment]
I think this needed to be [paused](https://expensify.slack.com/archives/C01SKUP7QR0/p1631574904303900) in favour of focusing on N6 projects. Added N6 hold the label. adding a weekly label to work on this in a week.

**Comment by Christinadobrzyn** [comment]
I think this needed to be paused in favour of focusing on N6 projects. Added N6 hold the label. adding a weekly label to work on this in a week.

Update: talking this over with Kadie, I think I can create the job in Upwork just no PR deployment until after N6.

**Comment by Christinadobrzyn** [comment]
Hey @bondydaa can I have your opinion if this GH should remain separate from this?

https://github.com/Expensify/App/issues/5113

I also saw this was deployed, do you think it might be causing this issue? 

https://github.com/Expensify/App/issues/4507

**Comment by isagoico** [comment]
Issue reproducible during KI retests.

**Comment by MelvinBot** [comment]
@Christinadobrzyn Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by Christinadobrzyn** [comment]
Asked Kadie to review the other language GHs to see if they might overlap - decided this issue is different. 

Created job in Upwork 

Internal posting - https://www.upwork.com/ab/applicants/1440191403556229120/job-details
External posting - https://www.upwork.com/jobs/~01010f498566c1d09e

**Comment by MelvinBot** [comment]
Triggered auto assignment to @marcaaron (`Exported`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by parasharrajat** [comment]
https://github.com/Expensify/App/pull/4789 & https://github.com/Expensify/App/pull/5019 are the two PRs that implement this behaviour to preserve the Locale on Login/logout.

I think we are missing a check for the beta for both changes done here. `!Permissions.canUseInternationalization(betas)`

https://github.com/Expensify/App/blob/a9af6e0ebf91a4bccff8c9f7a15de667e88ff68e/src/libs/Navigation/AppNavigator/AuthScreens.js#L172-L182

& 
https://github.com/Expensify/App/blob/a9af6e0ebf91a4bccff8c9f7a15de667e88ff68e/src/libs/actions/SignInRedirect.js#L42

If I am correct, both of code blocks should only run when Beta `!Permissions.canUseInternationalization(betas)` is true.

**Comment by iwiznia** [comment]
> If I am correct, both code blocks should only run when Beta !Permissions.canUseInternationalization(betas) is true.

Yes. Someone removed the beta (incorrectly) and re-added it later and maybe they forgot to re-add it here.

**Comment by parasharrajat** [comment]
Just adding Beta check should be enough to fix this issue as this happening when you switch between accounts with or without beta access.

**Comment by Christinadobrzyn** [comment]
@marcaaron sorry to ask but am I correct that this is the [PR](https://github.com/Expensify/App/pull/5019) for this fix and it was deployed over 7 days ago? 

Do we need to wait for something else to be deployed/fixed/completed before we pay @parasharrajat and close this?

**Comment by isagoico** [comment]
Issue reproducible during KI retests.

**Comment by Christinadobrzyn** [comment]
still on hold for n6

**Comment by Christinadobrzyn** [comment]
Hey team! Just checking in on where we are with this - has a PR been created or are we waiting on https://github.com/Expensify/App/pull/5019 to be resolved first?

**Comment by parasharrajat** [comment]
#5019 is a PR and that was already merged.

I have a proposal https://github.com/Expensify/App/issues/5286#issuecomment-924532697 and https://github.com/Expensify/App/issues/5286#issuecomment-924920008

**Comment by kadiealexander** [comment]
Please refer to [this post](https://expensify.slack.com/archives/C01GTK53T8Q/p1633557259458600) for updated information on the `n6 hold`, we've raised the bonus to **$250** for all issues where a **PR is created** before the N6 hold is lifted.

**Comment by Christinadobrzyn** [comment]
On hold for n6

**Comment by parasharrajat** [comment]
PR is updated.

**Comment by mvtglobally** [comment]
Issue reproducible during KI retests.

**Comment by Christinadobrzyn** [comment]
not overdue - still working on the PR

**Comment by marcaaron** [comment]
What is the status of this issue? It seems like we are working on it in parallel, but I'm not sure. Should it be closed?

Going OOO for the next week so going to unassign myself for now.

**Comment by marcaaron** [comment]
@nickmurray47 do you know if this is at all different from this issue https://github.com/Expensify/App/issues/6133?

**Comment by nickmurray47** [comment]
I think they are one and the same. The bug was that we were calling `setLocale()`, which sets the NVP and onyx key instead of just `Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, value)`. Testing from my PR fixed this bug so I think we should be good once it gets deployed.

**Comment by parasharrajat** [comment]
Ok. Thanks for letting us know @nickmurray47. I think we are good to close the PR then. @Christinadobrzyn Please suggest further action for this issue.

**Comment by Christinadobrzyn** [comment]
> I think they are one and the same. The bug was that we were calling setLocale(), which sets the NVP and onyx key instead of just Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, value). Testing from my PR fixed this bug so I think we should be good once it gets deployed.

@nickmurray47 - to clarify the next steps for this issue - can you confirm this Upwork job should be closed prior to the [PR](https://github.com/Expensify/App/pull/5943) going to production and @parasharrajat paid out?

**Comment by Christinadobrzyn** [comment]
Confirmed with @nickmurray47 to close this job. 

Paying @parasharrajat $250 for the proposal and the promised $250 bonus for the offsite hold and closing the job/this post.
